### DISCUSSION TITLE
How can i open "pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object"

### DISCUSSION THREAD

**Discussion by idkeverythingidk** [ansys/pyaedt#5169] [Q&A]
import pyaedt

hfss= pyaedt.Hfss()

oproject = hfss.odesktop.GetActiveProject()
odesign = oproject.GetActiveDesign()
omodule = hfss.odesign.GetModule("ReportSetup")
sol_data = omodule.GetSolutionDataPerVariation(
            "Modal Solution Data",

            "Setup1: Sweep",
            [
                "Domain:=", "Sweep"
            ],
            [
                "Freq:=", ["All"],
                "frame:=", ["All"]
            ],
            [
                "S(bs_1_rx_01_p1,bs_1_tx_01_p1)"
            ]
        )

####
print(sol_data)
[<pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6AFE50>, <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4050>, <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4250>, <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4450>, <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4650>, <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4850>, <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4A50>, <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4C50>]

Q. 
I swept freq 128 times and frame 8 times.
I think a 128 x 8 data table was created like the report.
But I don't know how to access these data.
Is there any way to do this?
This is really urgent.
Please help me.
Thanks for reading.

**Comment by Samuelopez-ansys** [comment]
@idkeverythingidk Please share the full code or a project (in a zip file) to reproduce the issue. The commands you used are AEDT API, it is not PyAEDT. If you are not using PyAEDT commands, and you prefer AEDT API, please submit a support case in the Ansys customer portal.

If you prefer PyAEDT API, you could try hfss.post.get_solution_data(...)

You can find some examples here:

https://aedt.docs.pyansys.com/version/stable/examples/05-Q3D/Q2D_Example_Stripline.html#create-setup-analyze-and-plot

https://aedt.docs.pyansys.com/version/stable/examples/02-SBR%2B/SBR_Example.html#plot-results-outside-aedt

**Reply by idkeverythingidk** [reply]
It doesn't matter which example result.
I just want to get the s-parameter result (Solution Data) as an array.
1. After using create_report (as a data table), what should I do after doing get_solution_data?
I don't know what this file format is <pyaedt.generic.grpc_plugin_dll_class.AedtObjWrapper object at 0x00000287DF6A4C50>.
2. Can I just use get_solution_per_variation to get it?

It doesn't matter which example,
Can you tell me how to get the solution data for all sweeps?
Thanks for your help.

**Reply by idkeverythingidk** [reply]
sol_data = omodule.GetSolutionDataPerVariation(
            "Modal Solution Data",

            "Setup1: Sweep",
            [
                "Domain:=", "Sweep"
            ],
            [
                "Freq:=", ["All"],
                "frame:=", ["All"]
            ],
            [
                "S(ant_veh_23_1_p1,ant_veh_24_1_p1)"
            ]
        ) 
-> spv=hfss.post.get_solution_data_per_variation(
    solution_type="Modal Solution Data",
    setup_sweep_name= "Setup1: Sweep",
    sweeps = {'Freq':'All', 'frame':'All'},
    expressions="S(ant_veh_24_1_p1,ant_veh_23_1_p1)"
    
)

Is it okay to apply it like this?
When I only use get_solution_data(),
only the sweep value for freq comes out, and the sweep value for frame does not come out.

**Reply by idkeverythingidk** [reply]
spv=hfss.post.get_solution_data_per_variation(
solution_type="Modal Solution Data",
setup_sweep_name= "Setup1: Sweep",
sweeps = {'Freq':'All', 'frame':'All'},
expressions="S(ant_veh_24_1_p1,ant_veh_23_1_p1)"

)
print(spv.variations)

[OrderedDict([('frame', 1.0)]), OrderedDict([('frame', 2.0)]), OrderedDict([('frame', 3.0)]), OrderedDict([('frame', 4.0)]), OrderedDict([('frame', 5.0)]), OrderedDict([('frame', 6.0)]), OrderedDict([('frame', 7.0)]), OrderedDict([('frame', 8.0)])]

However, when I try to access the data, I see that only the data for frame=1 is displayed.

dimg=spv.data_imag() -> When I changed it to numpy, I can still only get an array with 128 (the number of freq sweeps). How can I get data for the remaining frames 2 to 8?

**Reply by Samuelopez-ansys** [reply]
@idkeverythingidk If you do dir(spv) you will see how the object works.

You can even change the data to pandas format, spv.enable_pandas_output = False.

To change the variation, just modify it:
spv.active_variation = ...

**Reply by idkeverythingidk** [reply]
sol_data = hfss.post.get_solution_data_per_variation(
            solution_type="Modal Solution Data",
            setup_sweep_name="Setup1: Sweep",
            sweeps={'Freq': 'All', 'frame': 'All'},
            expressions="S(ant_veh_24_1_p1,ant_veh_23_1_p1)"
        ) 
print(dir(sol_data))


 ['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_convert_list_to_SI', '_enable_pandas_output', '_expressions', '_get_variations', '_ifft', '_init_solution_data_imag', '_init_solution_data_mag', '_init_solution_data_phase', '_init_solution_data_real', '_intrinsics', '_nominal_variation', '_original_data', '_primary_sweep', '_quantity', '_solutions_imag', '_solutions_mag', '_solutions_phase', '_solutions_real', '_sweeps_names', '_variation_tuple', 'active_expression', 'active_intrinsic', 'active_variation', 'data_db10', 'data_db20', 'data_imag', 'data_magnitude', 'data_phase', 'data_real', 'enable_pandas_output', 'export_data_to_csv', 'expressions', 'full_matrix_mag_phase', 'full_matrix_real_imag', 'ifft', 'ifft_to_file', 'init_solutions_data', 'intrinsics', 'is_real_only', 'nominal_variation', 'number_of_variations', 'plot', 'plot_3d', 'primary_sweep', 'primary_sweep_values', 'primary_sweep_variations', 'set_active_variation', 'to_degrees', 'to_radians', 'units_data', 'units_sweeps', 'update_sweeps', 'variation_values', 'variations']

Q.
If I change active variation to frame here, can I get all the values ​​of frequency for all frames?

I want to get the data table values ​​for all frames for all frequencies that I created with create_report as an array.

With the method you provided, 

>>>sol_data.active_variation="frame"


I'm really sorry, but I still don't understand what this module format is.
<pyaedt.modules.solutions.SolutionData object at 0x00000287E0D3D990>

**Reply by idkeverythingidk** [reply]
In the data table of the report, it is well expressed in the form of a+bj, but when using sol_data.full_matrix_real_imag, it comes out as (1.0, 27.99244): 5.304360373e-05.
(frame,freq):value, but the value comes out only as real, not complex. How can this problem be solved?

**Reply by Samuelopez-ansys** [reply]
Hi @idkeverythingidk,

In this example you can find more info:
https://aedt.docs.pyansys.com/version/stable/examples/07-Circuit/Circuit_AMI.html#get-ami-report

You should use an IDE like PyCharm or Visual Studio to analyze the variables, as you can see the "full_matrix_real_ima" returns 2 elements, real and imag part:

![image](https://github.com/user-attachments/assets/fc6fecd4-d2d9-4e84-bd74-43e7a0bcdd5a)

if you enable pandas, it is the same but you will obtain 2 DataFrames:

![image](https://github.com/user-attachments/assets/2dc426fe-59fb-45a5-b5b7-c7d6830406ff)
